### PR TITLE
Get emoji data from native

### PR DESCRIPTION
**Pull request by pederjohnsen**
Adds function to get emoji data from native unicode emoji

This resolves https://github.com/missive/emoji-mart/issues/16 and resolves https://github.com/missive/emoji-mart/issues/193

**[edit]**
Thought I had already fixed it but seems the issues with some emojis as mentioned in https://github.com/missive/emoji-mart/issues/16#issuecomment-390960784 are still present here too.. If anyone has any thoughts as to why and how to fix it let me know :upside_down_face:

### CODE REVIEW

**Comment by ifiokjr** [comment]
This looks very helpful. Any chance it gets merged soon?

**Comment by nolanlawson** [comment]
It seems based on discussion in #16 that this still needs some more work?

**Comment by pederjohnsen** [comment]
Yes unfortunately it still needs more work, I've not got the time to revisit this at the moment, but once I have (unless someone else has had a look by then) I'll post an update in here :ok_hand:

**Comment by pederjohnsen** [comment]
@nolanlawson I've updated the code and it seems to work correctly for all emojis now...

This also includes an update to `nimble-emoji-index.js` which returns search results for the skin tone you've picked in the picker. If I remember correctly this isn't required for the `getEmojiDataFromNative` to work. But was another thing I needed for an app I'm working on. I could pull this out and create a separate PR if wanted :+1:

File: src/utils/index.js [hunk 1]
```diff
@@ -134,6 +135,49 @@ function getData(emoji, skin, set, data) {
...
+  }
+
+  const skinTones = ['', '🏻', '🏼', '🏽', '🏾', '🏿']
+  const skinCodes = ['', '1F3FB', '1F3FC', '1F3FD', '1F3FE', '1F3FF']
+
+  let skin
+  let skinCode
+  let baseNativeString = nativeString
+
+  skinTones.forEach((skinTone) => {
```
Comment (nolanlawson) [inline src/utils/index.js]: Since the first `skinTone` is an empty string, is a good idea to search for it here?

File: src/utils/index.js [hunk 2]
```diff
@@ -134,6 +135,49 @@ function getData(emoji, skin, set, data) {
...
+    if (nativeString.indexOf(skinTone) > 0) {
+      const skinToneIndex = skinTones.indexOf(skinTone)
```
Comment (nolanlawson) [inline src/utils/index.js]: We can figure out this `skinToneIndex` by using `skinTones.forEach((skinTone, i) => {}) instead

File: src/utils/index.js [hunk 3]
```diff
@@ -134,6 +135,49 @@ function getData(emoji, skin, set, data) {
...
+      skin = skinToneIndex + 1
+      skinCode = skinCodes[skinToneIndex]
+    }
+  })
+
+  const emojiData = Object.values(data.emojis).find((emoji) => {
+    emoji = JSON.parse(_JSON.stringify(emoji))
```
Comment (nolanlawson) [inline src/utils/index.js]: Let's consistently use `_JSON` here

File: src/utils/index.js [hunk 4]
```diff
@@ -134,6 +135,49 @@ function getData(emoji, skin, set, data) {
...
+
+  skinTones.forEach((skinTone) => {
+    if (nativeString.indexOf(skinTone) > 0) {
+      const skinToneIndex = skinTones.indexOf(skinTone)
+      skin = skinToneIndex + 1
+      skinCode = skinCodes[skinToneIndex]
+    }
+  })
+
+  const emojiData = Object.values(data.emojis).find((emoji) => {
```
Comment (nolanlawson) [inline src/utils/index.js]: `Object.values()` is not supported in all browsers, this will probably require a new babel polyfill to work. Can we just use a for-loop instead?

File: src/utils/index.js [hunk 5]
```diff
@@ -134,6 +135,49 @@ function getData(emoji, skin, set, data) {
...
+
+    if (emoji.variations && emoji.variations.length) {
+        emoji.unified = emoji.variations.shift()
+    }
+
+    if (skin && emoji.skin_variations && emoji.skin_variations[skinCode]) {
+      emoji.unified = emoji.skin_variations[skinCode].unified
+    }
+
+    return unifiedToNative(emoji.unified) === baseNativeString
```
Comment (nolanlawson) [inline src/utils/index.js]: It seems odd to me to clone the whole object just to do a lookup, this seems expensive. Is there another way to do this?

File: .editorconfig [hunk 6]
```diff
@@ -0,0 +1,2 @@
+[*]
+indent_size = 2
```
Comment (nolanlawson) [inline .editorconfig]: This is a good change but would also probably make sense in a separate PR. :)

**Comment by pederjohnsen** [comment]
Thanks for reviewing, I'll try and get on this over the weekend 👍

**Comment by pederjohnsen** [comment]
@nolanlawson I've updated the code and added some tests, let me know if you find anything else :ok_hand:

**Comment by nolanlawson** [comment]
This LGTM, I think the only thing missing is that it would be great to have a Jest test for this, since this is a pretty self-contained little util function. I don't think it should necessarily block this PR from being merged, though.

**Comment by pederjohnsen** [comment]
https://github.com/missive/emoji-mart/pull/250/commits/ab2f6004ac5d99e3823630bfe6f98870205e99d2 ? :thinking:

**Comment by nolanlawson** [comment]
Oops, my bad, did not see the tests. Everything LGTM!

**Comment by pederjohnsen** [comment]
Thanks :) :tada:
